quarterly/2023-04-03-ledger.md

```diff
@@ -0,0 +1,355 @@
+  However, the ledger will reject a protocol parameter update on the epoch boundary
+  if the deposit pot adjustments cannot be reconciled with the reseve pot.
+  This means that if quorum is met regarding changing the major protocol version,
+  but the update is rejected on the epoch boundary, consensus will change the era but the
+  ledger will not change the major protocol version, leaving the ledger in a split-brain state.
+
+Because we never actually changed the values of the two deposits amounts in the protocol parameters
+on mainnet, we were able to retroactively change the behavior.
+We made the following changes:
+  * Individual deposits are tracked in the `DState`.
+  * The amount deposited is always returned.
+
+[[pull-3195]]
+[[pull-3202]]
+[[pull-3217]]
+
+#### New ledger API
+
+We have significantly built up the ledger API.
+We will eventually replace much of the `cardano-api` in the node repository with this ledger API.
+
+[[pull-3242]]
+[[pull-3248]]
+[[pull-3328]]
+
+#### Constraint-based generators
+
+Our largest scale property tests generate an initial ledger state and a long sequence of valid blocks
+which span several epochs, mimicking a real network.
+These tests are, in theory, excellent for checking properties.
+They are, however, very difficult to maintain and are not as random as we would like
+(a lot of bias has to be introduced to keep the ledger state in enough order to keep generating blocks).
+
+We have a new declaritive infrastructure for building constraint-based generators,
+which instead generate a random ledger state representative of not just an initial state,
+but also those representative of the end result of a long sequence of valid blocks.
+Moreover, these generators are very fast and are much more random than our old generators.
+Before we can start using them for our existing property tests, however, we still need to
+expand them to generate a valid block for a given ledger state.
+
+[[pull-3219]]
+
+#### Technical debt
+
+We continued to address technical debt as much as we can.
+
+[[pull-3167]]
+[[pull-3170]]
+[[pull-3172]]
+[[pull-3175]]
+[[pull-3184]]
+[[pull-3205]]
+[[pull-3208]]
+[[pull-3210]]
+[[pull-3212]]
+[[pull-3218]]
+[[pull-3222]]
+[[pull-3223]]
+[[pull-3224]]
+[[pull-3225]]
+[[pull-3229]]
+[[pull-3239]]
+[[pull-3241]]
+[[pull-3244]]
+[[pull-3245]]
+[[pull-3249]]
+[[pull-3260]]
+[[pull-3263]]
+[[pull-3264]]
+[[pull-3268]]
+[[pull-3269]]
+[[pull-3270]]
+[[pull-3274]]
+[[pull-3276]]
+[[pull-3277]]
+[[pull-3286]]
+[[pull-3290]]
+[[pull-3295]]
+[[pull-3296]]
+[[pull-3306]]
+[[pull-3307]]
+[[pull-3310]]
+[[pull-3311]]
+[[pull-3316]]
+[[pull-3320]]
+[[pull-3323]]
+[[pull-3327]]
+[[pull-3331]]
+[[pull-3332]]
+[[pull-3333]]
+[[pull-3338]]
+[[pull-3341]]
+[[pull-3347]]
+[[pull-3350]]
+[[pull-3351]]
+[[pull-3352]]
+[[pull-3354]]
+
+##### Critical fixes
+
+We fixed two critical issues:
+* Growing block production delay on the epoch boundary: [[pull-3209]]
```

Comment:
sadly, it seems we had to back this out at the last minute. :(

https://github.com/input-output-hk/cardano-ledger/pull/3375